src/core/on-chain/index.ts

```diff
@@ -80,9 +77,10 @@ export const OnChainMixin = (superclass) =>
         /// TODO: unable to check balanceSats vs this.dustThreshold at this point...
       }
 
-      return redlock({ path: this.user._id, logger: onchainLogger }, async (lock) => {
+      const walletId_ = this.user.walletId // FIXME: just set this variable for easier code review. long variable would trigger adding a tab and much bigger diff
+      return redlock({ path: walletId_, logger: onchainLogger }, async (lock) => {
         const balanceSats = await Wallets.getBalanceForWallet({
-          walletId: this.user.id,
+          walletId: this.user.walletId,
```

Comment:
The old user doesn't have a `walletId` does it?

Reply:
it does (with this PR)